src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -691,7 +689,7 @@ private function getProductURL(
             $ean13,
             $language->id,
             null,
-            (!$canonical) ? $product['id_product_attribute'] : null,
+            $product['id_product_attribute'],
```

Comment:
Is this the current product attribute?

Reply:
it is the current variant id depending on the context, it is always filled in the input array before present it (ie instanciating ProductLazyArray)

Reply:
Should it be the default combination selected in BO?

Reply:
you're right! my bad, I thought each combination had their own canonical, but it's the default one for all of them